src/rpc/net.cpp

```diff
@@ -200,6 +200,7 @@ static RPCHelpMan getpeerinfo()
         ServiceFlags services{fStateStats ? statestats.their_services : ServiceFlags::NODE_NONE};
         obj.pushKV("services", strprintf("%016x", services));
         obj.pushKV("servicesnames", GetServicesNames(services));
+        obj.pushKV("relaytxes", fStateStats ? statestats.m_relay_txs : false);
```

Comment:
A user reading `false` with bloom filters disabled may register the peer as one that txs aren't sent to, as the relay flag can't change after the handshake. So I think keeping this optional may be better. Also, in bitcoin-cli it might be better to display it as optional instead of a picking a constant which may later turn out to be wrong.

Reply:
> keeping this optional may be better

Note that this patch maintains the current API, whereas leaving it optional would annoyingly break user space without notice in v24. The API change was unclear but it appears unintended as it added neither optional mentions in the getpeerinfo help, which you fixed afterward, nor a release note. Absent a real need to change it (say, like removing a field), I have the impression that users prefer we don't break things willy-nilly, at least without some explanation of what and why (if we really, really want to make it optional ;).

Reply:
I think whether the change was "unintended" was discussed already, so I am not going to reply to that.

If there is a missing release note, it can be added to the wiki.

Reply:
To argue against my points, maybe getpeerinfo is mainly consumed by us developers and rarely by software clients or node operators -- I don't know -- and I agree with you that for us developers it is preferable not to return a relaytxes value while the connection is setting up. 

Updating to take your feedback (appreciated, thank you) with the caveats I described.

Reply:
> If there is a missing release note, it can be added to the wiki.

Note that it doesn't seem to have been added to the wiki if I'm parsing it correctly.

Reply:
I've also heard from the financial software industry using bitcoin that these v24 changes will break their production systems without a patch. It was first reported to me orally by an engineer who observed the discussions in #25923, and today I received written confirmation of this by PMs and engineers there.